frontend/src/features/drug_discovery/components/ligands/LigandsList.vue

```diff
@@ -0,0 +1,281 @@
+          console.error('Error deleting ligand:', error);
+        } finally {
+          this.$q.loading.hide();
+        }
+      },
+      toggleSelectAll(value: boolean) {
+        if (value) {
+          // Replace the array with a new one containing all IDs to ensure reactivity
+          this.selectedTargets = this.drugDiscoveryStore.targets.map(target => target.target_id);
+        } else {
+          // Clear the array by replacing it with a new empty array
+          this.selectedTargets = [];
+        }
+      },
+
+      async openTargetSelectionDialog(ligand: ExtendedLigandMetaData) {
+        this.selectedLigand = ligand;
+        this.targetSelectionDialogVisible = true;
+        this.selectedTargets = [];
+        this.selectAllTargets = false;
+        this.targets = await this.drugDiscoveryStore.fetchTargetsForExperiment(this.experimentId) as ExtendedTargetMetaData[];
+      },
+
+      async addLigandsToSelectedTargets() {
+        const store = useDrugDiscoveryStore();
+
+        for (let target_id of this.selectedTargets) {
+          if (this.selectedLigand && this.selectedLigand.data) {
+            // Assuming `selectedLigand.data` has the required SDF file or you have it available somehow
+            const sdfFileContent = this.selectedLigand.data.sdf_file;
+
+            const file = new File([new Blob([sdfFileContent!])], this.selectedLigand.ligand_name + ".sdf")
+
+            try {
+              await store.uploadLigandToTarget(this.experimentId, target_id.toString(), file);
+              // Handle success - Maybe show a notification?
+            } catch (error) {
+              console.error('Error uploading ligand to target:', error);
+              // Handle error - Show error message to user?
+            }
+          }
+        }
+
+        // Close dialog and clear selections after operations
+        this.targetSelectionDialogVisible = false;
+        this.selectedTargets = [];
+        this.selectAllTargets = false;
+      },
+      downloadLigand(ligand: ExtendedLigandMetaData) {
+        if (!ligand.data || !ligand.data.sdf_file) {
+          console.error('SDF file content is missing for this ligand.');
+          return;
+        }
+
+        // Create a blob from the SDF content
+        const sdfContent = ligand.data.sdf_file;
+        const blob = new Blob([sdfContent], { type: 'chemical/x-mdl-sdfile' });
+        const url = window.URL.createObjectURL(blob);
+
+        // Create a temporary anchor element and trigger the download
+        const a = document.createElement('a');
+        a.href = url;
+        a.download = `${ligand.ligand_name}.sdf`; // Set the file name
+        document.body.appendChild(a); // Append the anchor to the body
+        a.click(); // Trigger the download
+        window.URL.revokeObjectURL(url); // Clean up the URL object
+        a.remove(); // Remove the anchor from the document
+      },
+    },
+    unmounted() {
+      const bioBuddyStore = useBioBuddyStore();
```

Comment:
I would rather introduce a method in biobuddystore with unregister callback functionality. It will be used accross the solution, so we won't need to copy-paste the code below

Reply:
Yeah let's do after biobuddy refactoring, talked offline about it